### ISSUE TITLE
Update Example 

### ISSUE THREAD

**Issue by sadeghbarati** [kbrsh/moon#247]
hi @kbrsh 

pls add 0.10.0 examples to new moon like

`slash`
`wade`
`firebase`
`how to use external library like javascript or jQuery plugin in new moon`
`algolia search in new moon`

but in new moon example section we have one :sob: 

`todos`

**Comment by LeviSchuck** [comment]
It took me a moment, but it's really dang simple to introduce external events and APIs if you have

* Just 1 top level View component
* Just 1 data field driver thing

With the TODO's example, I was able to have it auto-add tasks every 3 seconds, a simulated event source.

```js
var eventNumber = 0;

setInterval(() => {
  console.log("Simulated event!");
  eventNumber += 1;
  Moon.run(({ data }) => {
    let newData = {...data}; // This isn't perfect for immutable data stuff, but you get the idea
    newData.todos.push("New Task " + eventNumber);
    return {
      data: newData,
      view: (<Todos data={data}/>)
    };
  });
}, 3000);
```

**Comment by kbrsh** [comment]
@LeviSchuck Cool!

I'd suggest moving `setInterval` to a driver though. The next beta has the ability to do this through the built-in `time` driver. It'll let you do this:

```js
const addTask = ({ data }) => ({
   data: {...data, todos: data.todos.concat(["New Task"])},
   time: { 3000: addTask }
});

Moon.run(({ data }) => ({
   view: (<Todos data={data}/>),
   time: { 3000: addTask }
}));
```

> This isn't perfect for immutable data stuff, but you get the idea

By the way, immutability is _encouraged_, not _required_. It's just to help make functions more concise and easier to understand, but you are free to do whatever you want, as long as you return a variable `data` for Moon to use as the new data.

**Comment by LeviSchuck** [comment]
I actually based this from the time code currently in the repository. My intent was to figure out how to make graphql requests, and to modify the data accordingly. Still working on that.

I have yet to grasp fully what drivers mean in the new moon, though the minimal source code makes it easier to make progress on my own.

**Comment by kbrsh** [comment]
Glad to see the source is a bit easier to understand — drivers are just objects with `input` and `output` functions. `input` provides data from outside sources (maybe a graphQL result), and `output` takes a value and does some effect with it.

A GraphQL driver might look like:

```js
let resultGlobal;

const GraphQL = {
   input() {
      return resultGlobal;
   },
   output({ query, after }) {
      fetchGraphQL(query).then(result => {
         resultGlobal = result;
         Moon.run(after);
      });
   }
};

Moon.use({
   GraphQL
});

Moon.run(() => ({
   GraphQL: {
      query: "{ query content here }",
      after: ({ GraphQL }) => {
         console.log(GraphQL);
         return {};
      }
   }
}));
```

**Comment by LeviSchuck** [comment]
Thank you! I've created a successful example.
It queries Pikachu's moves from the public pokemon demo graphql api, modifies the data, re-renders.

https://github.com/LeviSchuck/moon-graphql

![pokemons](https://user-images.githubusercontent.com/245911/63476289-04469480-c446-11e9-8e6b-81899c426f52.gif)
